### PR TITLE
Improved documentation for HTTP.Form and HTTP.Multipart.

### PR DESCRIPTION
**Pull request by fredrikekre**
xref e.g. https://stackoverflow.com/questions/66268254/julia-http-post-request-with-image-attachment

### CODE REVIEW

File: src/multipart.jl [hunk 1]
```diff
@@ -126,12 +157,11 @@ mutable struct Multipart{T <: IO} <: IO
     data::T
     contenttype::String
     contenttransferencoding::String
-    name::String
 end
 
-function Multipart(f::Union{AbstractString, Nothing}, data::T, ct::AbstractString="", cte::AbstractString="", name::AbstractString="") where {T<:IO}
```
Comment (fredrikekre) [inline src/multipart.jl:132]: `name` seems unused (?) and possibly a relic of old code where `Multipart` was not the value in `Form`?

Comment (quinnj) [reply src/multipart.jl:132]: Yeah, that sounds right

Comment (quinnj) [reply src/multipart.jl:132]: Oh, but we do have some tests that rely on it, which are failing here

Comment (fredrikekre) [reply src/multipart.jl:132]: Yea it seems that parsing a multipart body uses the name field. Perhaps parsing should give back `Vector{Pair{name::String,Multipart}}` instead to mirror how you construct a `HTTP.Form`, but I will leave that for another day and just leave the `name` field here.

**Pull request merged**